app/services/manage_donation.rb

```diff
@@ -1,4 +1,40 @@
-class ManageDonation < ManageAction
+class ManageDonation
+  def self.create(params)
+    new(params).create
+  end
+
+  def self.format_params(user_email:, user_country:, page_id:, amount:, exp_date:, card_num: '4111111111111111', additional_options = {})
```

Comment:
Unfortunately, with named parameters, you can't have a catch all hash ( `additional_options = {}` ). Are the additional options necessary?